python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -993,6 +843,180 @@ def _raise_on_error(self) -> None:
             )
 
 
+T = TypeVar("T", bound=DbtDagsterEventType)
+
+
+class DbtEventIterator(abc.Iterator[T]):
```

Comment:
I forgot that we could do this. I'm now very onboard with chaining methods after `.stream()`!